forest/oak/base.py

```diff
@@ -414,6 +415,130 @@ def find_continuous_dominant_peaks(valid_peaks: np.ndarray, min_t: int,
     return cont_peaks[:, :-1]
 
 
+def preprocess_dates(
+    file_list: list, time_start: Optional[str], time_end: Optional[str],
+    fmt: str, from_zone: Optional[tzinfo], to_zone: Optional[tzinfo]
+) -> tuple:
+    """Preprocesses dates of accelerometer files.
+
+    Args:
+        file_list: list of strings
+            list of accelerometer files
+        time_start: optional string
+            initial date of study in format: 'YYYY-mm-dd HH_MM_SS'
+        time_end: optional string
+            final date of study in format: 'YYYY-mm-dd HH_MM_SS'
+        fmt: string
+            format of dates in file_list
+        from_zone: tzinfo
+            timezone of dates in file_list
+        to_zone: tzinfo
+            local timezone
+    Returns:
+        Tuple of ndarrays:
+            - dates_shifted: list of datetimes with hours set to 0
+            - date_start: datetime of initial date of study
+            - date_end: datetime of final date of study
+    """
+    # transform all files in folder to datelike format
+    file_dates = [
+        file.replace(".csv", "").replace("+00_00", "")
+        for file in file_list
+    ]
+
+    # process dates
+    dates = [datetime.strptime(file, fmt) for file in file_dates]
+    dates = [
+        date.replace(tzinfo=from_zone).astimezone(to_zone)
+        for date in dates
+    ]
+
+    # trim dataset according to time_start and time_end
+    if time_start is not None and time_end is not None:
+        time_min = datetime.strptime(time_start, fmt)
+        time_min = time_min.replace(tzinfo=from_zone).astimezone(to_zone)
+        time_max = datetime.strptime(time_end, fmt)
+        time_max = time_max.replace(tzinfo=from_zone).astimezone(to_zone)
+        dates = [date for date in dates if time_min <= date <= time_max]
+
+    dates_shifted = [date-timedelta(hours=date.hour) for date in dates]
+
+    # create time vector with days for analysis
+    if time_start is None:
+        date_start = dates_shifted[0]
+        date_start = date_start - timedelta(hours=date_start.hour)
+    else:
+        date_start = datetime.strptime(time_start, fmt)
+        date_start = date_start.replace(
+            tzinfo=from_zone
+        ).astimezone(to_zone)
+        date_start = date_start - timedelta(hours=date_start.hour)
+
+    if time_end is None:
+        date_end = dates_shifted[-1]
+        date_end = date_end - timedelta(hours=date_end.hour)
+    else:
+        date_end = datetime.strptime(time_end, fmt)
+        date_end = date_end.replace(tzinfo=from_zone).astimezone(to_zone)
+        date_end = date_end - timedelta(hours=date_end.hour)
+
+    return dates_shifted, date_start, date_end
+
+
+def run_hourly(
+    t_hours_pd: pd.Series, days_hourly: pd.DatetimeIndex,
+    cadence_bout: np.ndarray, steps_hourly: np.ndarray,
+    walkingtime_hourly: np.ndarray, cadence_hourly: np.ndarray,
+    frequency: Frequency
+) -> None:
```

Comment:
This function does not return a value or write anything to files. What happens to the result of this computation?

Reply:
It updates the arrays, `steps_hourly`, `walkingtime_hourly` and `cadence_hourly`

Reply:
I see, those are mutable types. It would be great to mention in the docstring that this function modifies those arguments in place.